### PR TITLE
Extension store base class + rename main/renderer-extensions -> extensions

### PR DESCRIPTION
**Pull request by jakolehm**
Implements safer base class for extension stores. While implementing this I noticed huge problems how we did export types. If same type is exported through different module names they are not exactly the same type -> compile fails no matter what user does. ~~This is the reason why I had to refactor `main-extensions` -> `core-extensions`  and move common logic/types there.~~ This is the reason why I had to refactor everything to a single package. 

I really hope this is the last package naming change here ...

### CODE REVIEW

File: extensions/example-extension/page.tsx [hunk 1]
```diff
@@ -1,13 +1,14 @@
-import { Component, LensRendererExtension } from "@k8slens/renderer-extensions";
+import { LensRendererExtension } from "@k8slens/core-extensions";
+import { Component } from "@k8slens/renderer-extensions";
```
Comment (ixrock) [inline extensions/example-extension/page.tsx]: Import of `LensRendererExtension` still comes from `core` package but keeps name about renderer..
Not too obvious when there are many packages from `@k8slens/*`

Maybe we could make single package import like `@k8slens/extension` for both `main` and `render` processes and provide all types from there, but depending on process not everything would be available in runtime (just like `electron` serving everything from single package, `app`/`ipcMain`/etc. only for main but from same pack).

Comment (jakolehm) [reply extensions/example-extension/page.tsx]: @ixrock I like this idea... let's see if we can make it work.

**Review by ixrock — APPROVED** [review]
Looking good!
Only one thing I would change: `core-*` names replace with `main-*` to be more aligned with `renderer-*` file/class names.

**Comment by jakolehm** [comment]
> Only one thing I would change: core-* names replace with main-* to be more aligned with renderer-* file/class names.

Let's figure this out later (it does not affect public api).

**Pull request merged**